src/main/java/com/salesforce/dataloader/model/LoginCriteria.java

```diff
@@ -96,7 +96,6 @@ public void setSessionId(String sessionId) {
     public void updateConfig(Config config) {
         config.setValue(Config.USERNAME, config.STRING_DEFAULT);
         config.setValue(Config.PASSWORD, config.STRING_DEFAULT);
-        config.setValue(Config.ENDPOINT, config.STRING_DEFAULT);
```

Comment:
Jae, see my overall comment above. It is a little bit complex. More specific details: with setting the endpoint to empty string(STRING_DEFAULT), it will update the config. Then the next time Setting dialog or log in dialog with local a null string, then it will use a default string from partner API. With the fix, will keep the last good endpoint, so you can still login correctly. :-)